File: works/tmp/1912.cs
using System;
using System.Collections;
using static System.Console;

namespace ConsoleApp
{
    class Program
    {

        static void Main(string[] args)
        {
            int n = int.Parse(ReadLine());
            int[] a = new int[100001];
            int[] d = new int[100001];

            string[] _s = ReadLine().Split();
            for (int i = 0; i < n; i++) a[i] = int.Parse(_s[i]);

            d[0] = a[0];
            int answer = d[0];
            for (int i = 1; i < n; i++)
            {
                d[i] = Math.Max(d[i - 1] + a[i], a[i]);
                answer = Math.Max(answer, d[i]);
            }
            Write(answer);
        }
    }

}

File: works/tmp/1912.cpp
#include <iostream>
using namespace std;

int main()
{
    int n;
    int a[100001] = {0,};
    int d[100001] = {0,};
    cin >> n;
    for(int i = 0; i < n; i++)
    {
        int num; cin >> num;
        a[i] = num;
    }
    
    d[0] = a[0];
    int answer = d[0];
    for(int i = 1; i < n; i++)
    {
        d[i] = max(d[i-1]+a[i], a[i]);
        answer = max(answer, d[i]);
    }

    cout << answer;
}